src/ng/directive/validators.js

```diff
@@ -68,15 +68,21 @@ var requiredDirective = ['$parse', function($parse) {
     require: '?ngModel',
     link: function(scope, elm, attr, ctrl) {
       if (!ctrl) return;
-      var value = attr.required || $parse(attr.ngRequired)(scope);
+      // For boolean attributes like required, presence means true
+      var value = 'required' in attr || $parse(attr.ngRequired)(scope);
```

Comment:
Nit: `attr.hasOwnPropetty('required')` would avoid looking up the prototype chain and be more consistent with similar checks in other places.